backend/wayland/wl_seat.c

```diff
@@ -8,34 +8,188 @@
 #include <wlr/interfaces/wlr_input_device.h>
 #include <wlr/interfaces/wlr_pointer.h>
 #include <wlr/interfaces/wlr_keyboard.h>
+#include <wlr/interfaces/wlr_touch.h>
 #include <wlr/util/log.h>
 #include "backend/wayland.h"
 
-static void wlr_wl_device_destroy(struct wlr_input_device_state *state) {
+static void pointer_handle_enter(void *data, struct wl_pointer *wl_pointer,
+		uint32_t serial, struct wl_surface *surface, wl_fixed_t surface_x,
+		wl_fixed_t surface_y) {
+	struct wlr_input_device *dev = data;
+	assert(dev && dev->pointer && dev->pointer->state);
+	struct wlr_pointer_state *state = dev->pointer->state;
+	state->surface_x = wl_fixed_to_double(surface_x);
+	state->surface_y = wl_fixed_to_double(surface_y);
+}
+
+static void pointer_handle_leave(void *data, struct wl_pointer *wl_pointer,
+		uint32_t serial, struct wl_surface *surface) {
+
```

Comment:
Could destroy the input device and recreate it when it re-enters. Would cause compositors to remove the cursor. Not a blocker but food for thought.